src/main/kotlin/frc/robot/subsystems/climb/Climb.kt

```diff
@@ -1,65 +1,54 @@
 package frc.robot.subsystems.climb
 
-import edu.wpi.first.wpilibj.Timer
 import edu.wpi.first.wpilibj2.command.Command
 import edu.wpi.first.wpilibj2.command.Commands
 import edu.wpi.first.wpilibj2.command.SubsystemBase
-import org.littletonrobotics.junction.AutoLogOutput
 import org.littletonrobotics.junction.Logger
 import java.util.function.DoubleSupplier
 import kotlin.math.absoluteValue
 
 class Climb private constructor(private val io: ClimbIO) : SubsystemBase() {
     private val inputs = io.inputs
-    private val timer = Timer()
 
-    @AutoLogOutput
-    private var isStopperStuck = false
+    private var isStopperStuck: Boolean = false
 
     companion object {
         @Volatile
         private var instance: Climb? = null
 
         fun initialize(io: ClimbIO) {
             synchronized(this) {
-                if (instance == null)
+                if (instance == null) {
                     instance = Climb(io)
+                }
             }
         }
 
         fun getInstance(): Climb {
-            return instance ?: throw IllegalStateException("climb not initialized, call initialize(io)")
+            return instance ?: throw IllegalStateException(
+                "Climb has not been initialized. Call initialize(io: ClimbIO) first."
+            )
         }
     }
 
-    init {
-        timer.start()
-        timer.reset()
-    }
-
-    fun open(): Command {
-        return Commands.run(io::openStopper)
-            .until { isStopperStuck }
-            .andThen(io::disableStopper)
+    fun setPower(power: DoubleSupplier): Command {
+        return Commands.run({ io.setPower(power.asDouble) }).withName("setPower")
     }
 
     fun lock(): Command {
-        return Commands.run(io::closeStopper)
-            .until { isStopperStuck }
-            .andThen(io::disableStopper)
+        return Commands.runOnce({ io.lockClimb() }).until(::isStopperStuck).andThen({ io.disableLockMotor() })
```

Comment:
Use method referencing, and there is no need for regular braces you can just keep only the curly ones in.
Same in unlock method aswell.